### DISCUSSION TITLE
Wix v4 - Compiles fine and the MSIs are running fine, But the Bundle EXE is Not Executing as it is not showing anything.

### DISCUSSION THREAD

**Discussion by garevaul** [wixtoolset/issues#7349] [Questions] [answered]
Hi,

I upgraded my Wix3 projects to Wix v4 using Heatwave.

I have 3 Projects in the Solution. 
1) X86 MSI 
2) X64 MSI 
3) Bundle EXE

The Bundle EXE will use the OS Relevant MSI to Install.

The Solution Compiles Fine. 

The MSI's are working fine.

But the EXE Does not Show me anything. Seems it is not Running at all.

FYI: I added the bal:DisplayInternalUICondition="yes" to the MSIPackage Elements for use in Wix v4. It compiles fine, but maybe it does not work this way.

Regards

This is the Bundle.wxs code

```
<Wix xmlns="http://wixtoolset.org/schemas/v4/wxs" xmlns:bal="http://wixtoolset.org/schemas/v4/wxs/bal" xmlns:util="http://wixtoolset.org/schemas/v4/wxs/util">

	<Bundle Name="!(loc.BundleName)" Version="0.0.1" Manufacturer="!(loc.ManufacturerName)" UpgradeCode="!(loc.UpgradeCode)" IconSourceFile="!(loc.IconSourceFile)" UpdateUrl="!(loc.UpdateUrl)" AboutUrl="!(loc.AboutUrl)" HelpUrl="!(loc.HelpUrl)" Copyright="!(loc.Copyright)" Compressed="yes" HelpTelephone="!(loc.HelpTelephone)" DisableModify="yes">

		<?define MSIUpgradeCode = {09427168-8FE3-45C1-B920-9380BEEB644C} ?>
		<?define ProductNameAndVersion = GoodbyteCoderMaster 0.0.1 ?>
		<?define ProductVersion = 0.0.1 ?>

		<BootstrapperApplication>
			<bal:WixStandardBootstrapperApplication LicenseUrl="" ThemeFile="HyperlinkTheme.xml" LocalizationFile="HyperlinkTheme.wxl" SuppressOptionsUI="yes" LogoFile="GB_Logo_65_by_65.png" Theme="hyperlinkLicense" />
		</BootstrapperApplication>

		<Chain>
			<MsiPackage bal:DisplayInternalUICondition="yes" InstallCondition="VersionNT64" SourceFile="$(var.OurApplicationsSetupX64.TargetDir)!(loc.MSIFileNameX64)"  Cache="keep" >
			</MsiPackage>
			<MsiPackage bal:DisplayInternalUICondition="yes" InstallCondition="NOT VersionNT64" SourceFile="$(var.OurApplicationsSetup.TargetDir)!(loc.MSIFileNameX86)" Cache="keep" >
			</MsiPackage>
		</Chain>
	</Bundle>
</Wix>

```

FYI : This is the Candle and Light Version I have on my computer (If that makes a difference)

![image](https://user-images.githubusercontent.com/15125247/229336532-d51b84d4-c5e7-40a8-9217-9474882b9493.png)

![image](https://user-images.githubusercontent.com/15125247/229336596-18a666b8-ec17-43f4-ad19-ab5d09ed5044.png)

And the Solution Configuration Manager is ;

![image](https://user-images.githubusercontent.com/15125247/229355693-de500808-2651-4ae7-ade8-88de4bc2d950.png)

And there are Multiple 'Debug' Folders;
1. 'obj\Debug', 'bin\Debug'
2. 'obj\x86\Debug', 'bin\x86\Debug'

![image](https://user-images.githubusercontent.com/15125247/229357346-deddab02-4520-44d1-b3c0-8cf935da2b4b.png)

![image](https://user-images.githubusercontent.com/15125247/229357413-ac61a589-0a77-4a0f-a6c7-ced3863db94b.png)

**Comment by nirbar** [comment]
DisplayInternalUICondition is a condition. To always show the MSI UI, use something like "1=1" or "not never"

**Comment by nirbar** [comment]
The EXE doesn't show? Do you mean that the main bundle doesn't start? Or the MSI UI?
Please attach log files

**Reply by garevaul** [reply]
@nirbar 

The Exe does not Execute. The MSI's Individually Execute Fine.

Here are the Log files of the Exe Execution.

[Goodbyte Coder Master 0.0.1_20230402201036.cleanroom.log](https://github.com/wixtoolset/issues/files/11132487/Goodbyte.Coder.Master.0.0.1_20230402201036.cleanroom.log)
[Goodbyte Coder Master 0.0.1_20230402201037.log](https://github.com/wixtoolset/issues/files/11132488/Goodbyte.Coder.Master.0.0.1_20230402201037.log)

**Comment by nirbar** [comment]
The log show failure on image id:
```
Failed to find image id
```
It isn't related to the MSI internal UI

**Reply by garevaul** [reply]
I think it does not say which image file. There is no name mentioned.

How can I know which image file ?

Regards

**Reply by rseanhall** [reply]
The problem is in your theme file (`HyperlinkTheme.xml`). You are probably trying to use your v3 theme which won't work.

**Reply by garevaul** [reply]
@rseanhall any suggestions on how I can replace it with a v4 theme file. Any URL to the file.

Or maybe if I create a new v4 project using the Wix v4 Heatwave template, then maybe the theme file will be there in the project  ?

Regards

**Reply by robmen** [reply]
FYI, HeatWave does not have a template for this... yet.

**Reply by garevaul** [reply]
Ok @robmen, but how can i replace the v3 theme file with a v4 one. Where is it ?

Regards

**Reply by rseanhall** [reply]
You will want to read https://wixtoolset.org/docs/fourthree/faqs/. If HeatWave is not pointing users to that webpage after upgrading a bundle project then it probably should be updated to do that.

**Comment by garevaul** [comment, marked as answer]
@rseanhall @robmen @nirbar 

Thanx to you guys. It works now.

I am posting this, so it will benefit others who see this article.

1) Firstly <MsiPackage bal:DisplayInternalUICondition="1=1".... as suggested by @nirbar works Good.

2) Secondly the link https://wixtoolset.org/docs/fourthree/faqs/ provided by @rseanhall leads to an article which has the link https://github.com/wixtoolset/wix4/tree/develop/src/ext/Bal/wixstdba/Resources which contains the HyperlinkTheme.xml & HyperlinkTheme.wxl files which I incorporated in my Upgraded Wix4 Bundle Project, by making some changes in accordance with what I had implemented in my Wix3 Project.

Now it is working fine.

Regards
@garevaul
